apps/studio/pages/api/ai/sql/generate-v4.ts

```diff
@@ -74,7 +67,17 @@ async function handlePost(req: NextApiRequest, res: NextApiResponse) {
     return res.status(400).json({ error: 'Invalid request body', issues: parseError.issues })
   }
 
-  const { messages, projectRef, connectionString, orgSlug, chatName } = data
+  const { messages: rawMessages, projectRef, connectionString, orgSlug, chatName } = data
+
+  // Server-side safety: limit to last 5 messages and remove `results` property to prevent accidental leakage
+  const messages = (rawMessages || []).slice(-5).map((msg: any) => {
+    if (msg && msg.role === 'assistant' && 'results' in msg) {
+      const cleanedMsg = { ...msg }
+      delete cleanedMsg.results
```

Comment:
What is the negative impact of this? Every new round of messages loses results from the previous set?